### PR TITLE
feat(cli): move stacks in new files for fresh projects

### PR DESCRIPTION
**Pull request by DanielMSchmidt**

### CODE REVIEW

File: packages/cdktf-cli/templates/java/src/main/java/com/mycompany/app/MainStack.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.mycompany.app;
+
+import software.constructs.Construct;
+
+import com.hashicorp.cdktf.TerraformStack;
+
+public class Main extends TerraformStack
```
Comment (ansgarm) [inline packages/cdktf-cli/templates/java/src/main/java/com/mycompany/app/MainStack.java:7]: Should we name this "MyStack" to be similar to how our TypeScript template looks?

Comment (jsteinich) [reply packages/cdktf-cli/templates/java/src/main/java/com/mycompany/app/MainStack.java:7]: Either `MyStack` or `MainStack` would be better than `Main`. Also need to make sure the class name matches the file name.

File: packages/cdktf-cli/templates/java/src/main/java/com/mycompany/app/Main.java [hunk 2]
```diff
@@ -5,17 +5,13 @@
...
 import com.hashicorp.cdktf.TerraformStack;
 
-public class Main extends TerraformStack
-{
-    public Main(final Construct scope, final String id) {
-        super(scope, id);
-
-        // define resources here
-    }
+import com.mycompany.app.MainStack;
```
Comment (jsteinich) [inline packages/cdktf-cli/templates/java/src/main/java/com/mycompany/app/Main.java:8]: This should be in the `com.mycompany.app` package and then this import won't be necessary.

File: packages/cdktf-cli/templates/csharp/MainStack.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using System;
+using Constructs;
+using HashiCorp.Cdktf;
+
+
+namespace MyCompany.MyApp
+{
+    class MyApp : TerraformStack
```
Comment (jsteinich) [inline packages/cdktf-cli/templates/csharp/MainStack.cs:8]: Should use the same class name for the stack as decided with the Java version. Make sure the filename matches up as well (not a hard requirement in C#, but good practice).

File: packages/cdktf-cli/templates/csharp/Main.cs [hunk 4]
```diff
@@ -5,13 +5,8 @@
 
 namespace MyCompany.MyApp
 {
-    class MyApp : TerraformStack
+    class MyApp
```
Comment (jsteinich) [inline packages/cdktf-cli/templates/csharp/Main.cs:8]: `Program` is the standard name for an entry point class.

**Review by jsteinich — CHANGES_REQUESTED** [review]

File: packages/cdktf-cli/templates/csharp/Main.cs [hunk 5]
```diff
@@ -5,13 +5,8 @@
 
 namespace MyCompany.MyApp
 {
-    class MyApp : TerraformStack
+    class MyApp : Program
```
Comment (jsteinich) [inline packages/cdktf-cli/templates/csharp/Main.cs:8]: Should probably rename the file to `Program.cs`. Will need to update hooks and some tests to account for that change.

**Review by jsteinich — CHANGES_REQUESTED** [review]

**Review by ansgarm — APPROVED** [review]
Nice!
We need to update the Learn tutorials for C# and Java after we released this and check our docs whether they mention the older way.
Let's add some kind of "post 0.12 release" issue to keep track of that.

File: packages/cdktf-cli/templates/csharp/Program.cs [hunk 6]
```diff
@@ -5,13 +5,8 @@
 
 namespace MyCompany.MyApp
 {
-    class MyApp : TerraformStack
+    class MyApp : Program
```
Comment (jsteinich) [inline packages/cdktf-cli/templates/csharp/Program.cs:8]: The class name should be `Program` and it won't extend/implement anything.

File: packages/cdktf-cli/templates/csharp/MainStack.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
+using System;
+using Constructs;
+using HashiCorp.Cdktf;
+
+
+namespace MyCompany.MyApp
+{
+    class MainStack : TerraformStack
```
Comment (jsteinich) [inline packages/cdktf-cli/templates/csharp/MainStack.cs:8]: Would be good to actually use this in the template.

File: packages/cdktf-cli/templates/csharp/Program.cs (hunk 6 above)
Comment (jsteinich) [reply packages/cdktf-cli/templates/csharp/Program.cs:8]: Also creating an instance of this below will fail since that constructor doesn't exist.

**Pull request merged**